src/platform/windows/display_vram.cpp

```diff
@@ -910,97 +920,228 @@ namespace platf::dxgi {
         BOOST_LOG(info) << "Capture format changed ["sv << dxgi_format_to_string(capture_format) << " -> "sv << dxgi_format_to_string(desc.Format) << ']';
         return capture_e::reinit;
       }
+    }
 
-      // Now that we know the capture format, we can finish creating the image
-      if (complete_img(img, false)) {
-        return capture_e::error;
-      }
+    enum class lfa {
+      nothing,
+      replace_surface_with_img,
+      replace_img_with_surface,
+      copy_src_to_img,
+      copy_src_to_surface,
+    };
 
-      // Copy the texture to use for cursor-only updates
-      device_ctx->CopyResource(last_frame_copy.get(), src.get());
+    enum class ofa {
+      forward_last_img,
+      copy_last_surface_and_blend_cursor,
+      dummy_fallback,
+    };
 
-      // Copy into the capture texture on the image with the mutex held
-      status = img->capture_mutex->AcquireSync(0, INFINITE);
-      if (status != S_OK) {
-        BOOST_LOG(error) << "Failed to acquire capture mutex [0x"sv << util::hex(status).to_string_view() << ']';
-        return capture_e::error;
+    auto last_frame_action = lfa::nothing;
+    auto out_frame_action = ofa::dummy_fallback;
+
+    if (capture_format == DXGI_FORMAT_UNKNOWN) {
+      last_frame_action = lfa::nothing;
+      out_frame_action = ofa::dummy_fallback;
+    }
+    else {
+      if (src) {
+        if (draw_mouse_cursor_flag) {
+          last_frame_action = lfa::copy_src_to_surface;
+          out_frame_action = ofa::copy_last_surface_and_blend_cursor;
+        }
+        else {
+          last_frame_action = lfa::copy_src_to_img;
+          out_frame_action = ofa::forward_last_img;
+        }
+      }
+      else if (!std::holds_alternative<std::monostate>(last_frame_variant)) {
+        if (draw_mouse_cursor_flag) {
+          if (std::holds_alternative<std::shared_ptr<platf::img_t>>(last_frame_variant)) {
+            last_frame_action = lfa::replace_img_with_surface;
+          }
+          out_frame_action = ofa::copy_last_surface_and_blend_cursor;
+        }
+        else {
+          if (std::holds_alternative<texture2d_t>(last_frame_variant)) {
+            last_frame_action = lfa::replace_surface_with_img;
+          }
+          out_frame_action = ofa::forward_last_img;
+        }
+      }
+      else {
+        last_frame_action = lfa::nothing;
```

Comment:
Is this path actually reachable? It seems like we should only hit the dummy fallback for `capture_format == DXGI_FORMAT_UNKNOWN`. For other cases, we should always have something to give the caller.